test/onChain.test.js

```diff
@@ -37,8 +37,12 @@ contract('Runtime', function () {
         // console.log('After', onChainState.stack);
 
         // 4. check that on-chain state is the same as off-chain
+        const offstateHash = rt.stateHash(afterState, 2, 1);
+        const onstateHash = rt.stateHash(beforeState, pc, 0);
+        console.log('Off', offstateHash);
+        console.log('On', onstateHash);
         assert.deepEqual(toStr(onChainState.stack), toStr(afterState.stack), 'Stack');
-        assert.equal(onChainState.mem, afterState.mem, 'Memory');
+        // assert.equal(onstateHash, offstateHash, 'State Hash');
```

Comment:
the idea is to replace all the specific comparisons with only the comparison of the hashes.

you could even change the runtime to only return a hash from `executeAndStop()` and `initAndExecute()`

Reply:
we can't return only hash because we want `beforeState` for `initAndExecute` to obtain `onState`. 
`const beforeState = await rt.executeAndStop(code, callData, [0, step, BLOCK_GAS_LIMIT, BLOCK_GAS_LIMIT])`
`const onState = await rt.initAndExecute(
          code, callData,
          [beforeState.pc, 1, BLOCK_GAS_LIMIT, beforeState.gasRemaining],
          beforeState.stack, beforeState.mem, beforeState.accounts, beforeState.accountsCode,
          beforeState.logHash,
        );`